### PR TITLE
Front end work v3

### PR DESCRIPTION
**Pull request by masoncpott**
@davidsjeong @danielharmon @gigicupp 

I did quite a bit of front end work on this. You can ignore the "IconsComponent.jsx" work, I have decided to cut a separate branch for that.

The highlights for this one:
-the product feature details are now dynamically rendering in roughly the correctly location and with the correct font.
-they need to be spaced out a little more but I will do that later this week.
-I changed a class based component to be a functional component.
-I added a "Materials" component in preparation for incorporating the material information from the materialsAPI. It is just a skeleton right now but it is in the correct location and has the correct styles on it.

Thank you!

### CODE REVIEW

File: client/src/components/ProductDetails.jsx [hunk 1]
```diff
@@ -41,10 +41,14 @@ class ProductDetails extends React.Component {
         <FeatureHeader>Features</FeatureHeader>
         <DetailsContainer>
           <DetailsRow>
-            <div>Feature Title</div>
-            <div>Feature Description</div>
-            <div>Feature Title</div>
-            <div>Feature Description</div>
+            <div>
+              <div>{this.props.details.feature1_title}</div>
```
Comment (gigicupp) [inline client/src/components/ProductDetails.jsx]: @masoncpott you could probably de-structure your this.props.details up top and use the variables here. Just my two cents.

File: client/src/components/IconComponent.jsx [hunk 2]
```diff
@@ -16,13 +23,13 @@ class IconComponent extends React.Component {
...
...5.491zm15.56 5.766c-1.3.453-2.036.713-2.451.86l.943 1.642c.28.512.33 1.085.16 1.61l1.367-.018c.266-.006 1.123-.127 1.147-1.321a4.558 4.558 0 0 0-.517-1.496zM15.945 3l.29.003c2.141.052 4.027 1.195 5.054 3.06l1.365 2.441 2.19-1.206a.699.699 0 0 1 1.022.747l-1.558 7.946a.698.698 0 0 1-.96.509l-7.206-3.061a.698.698 0 0 1-.07-1.252l1.974-1.118-1.372-2.454a.615.615 0 0 0-.532-.338.609.609 0 0 0-.575.302l-.914 1.6a2.341 2.341 0 0 1-.069.124l-.12.188a.971.971 0 0 1-.063.12.825.825 0 0 1-.555.388c-.175.033-.443.025-.735-.197-.203-.154-2.797-2.29-2.966-2.436a.699.699 0 0 1-.128-.906l1.052-1.632a5.941 5.941 0 0 1 1.308-1.488.693.693 0 0 1 .225-.201c.042-.023.082-.04.123-.06a5.861 5.861 0 0 1 3.106-1.07.684.684 0 0 1 .113-.009zM16 4.398c-.019.003-.038.004-.057.005-.058.002-.114.013-.171.017a2.5 2.5 0 0 1 .476.6l1.772 3.124c.014.024.022.05.032.075l1.553 2.775a.698.698 0 0 1-.266.95l-1.348.762 5.11 2.17 1.105-5.632-1.487.82a.699.699 0 0 1-.947-.272l-1.706-3.05a4.477 4.477 0 0 0-3.857-2.34zm-.97 1.306c-.13-.22-.62-.876-1.606-.415-.46.341-.859.77-1.17 1.275-.001.005-.004.009-.007.013a.564.564 0 0 1-.033.047l-.689 1.072c.55.455 1.454 1.2 1.995 1.641l.842-1.467a2.011 2.011 0 0 1 1.357-.95z"> </path>
```
Comment (gigicupp) [inline client/src/components/IconComponent.jsx:28]: just out of curiosity, what does this <path> tag do?

File: client/src/components/ProductDetails.jsx [hunk 3]
```diff
@@ -31,7 +31,21 @@ const DetailsRow = styled.div`
 // display all rows in a column
 
 function ProductDetails({ details }) {
-  const { feature1_title, feature1_description } = details;
+  const {
```
Comment (gigicupp) [inline client/src/components/ProductDetails.jsx:35]: nice!

**Review by gigicupp — APPROVED** [review]
Hey @masoncpott looks good. You have made a lot of progress!
Please ignore the first comment on de-structuring. You have done that in the later commits.
Awesome job

File: client/src/components/IconComponent.jsx (hunk 2 above)
Comment (masoncpott) [reply client/src/components/IconComponent.jsx:28]: @gigicupp this tag renders an SVG icon. It was a real pain to figure this out, i probably could have gotten away with using a library

**Pull request merged**